ding/framework/supervisor.py

```diff
@@ -199,12 +199,19 @@ class Supervisor:
         ChildType.THREAD: queue.Queue
     }
 
-    def __init__(self, type_: ChildType) -> None:
+    def __init__(self, type_: ChildType, subprocess_use_cuda: bool = False) -> None:
         self._children: List[Child] = []
         self._type = type_
         self._child_class = self.TYPE_MAPPING[self._type]
         self._running = False
         self.__queue = None
+        self._subprocess_use_cuda = subprocess_use_cuda
+        # If create a CUDA context before the fork(), you cannot use that within the
+        # child process. The key point is to not create a CUDA context before the fork.
+        # So use 'spawn' instead of 'fork', ref:
+        # https://stackoverflow.com/questions/22950047/cuda-initialization-error-after-fork
+        if subprocess_use_cuda:
+            Supervisor.QUEUE_MAPPING[ChildType.PROCESS] = mp.get_context('spawn').Queue
```

Comment:
maybe you should set global context by `mp.set_start_method`

Reply:
Done, use Supervisor's mp_ctx init arg to select 'spawn' context.